examples_tests/01.HelloWorld/main.cpp

```diff
@@ -108,85 +89,127 @@ class DemoEventCallback : public nbl::ui::IWindow::IEventCallback
 	{
 		LOG("A keyboard has been disconnected");
 	}
+
+public:
+	bool m_gotWindowClosedMsg = false;
 };
 
+static core::smart_refctd_ptr<system::ISystem> createSystem()
+{
+	core::smart_refctd_ptr<system::ISystemCaller> caller = nullptr;
+#ifdef _NBL_PLATFORM_WINDOWS_
+	caller = core::make_smart_refctd_ptr<nbl::system::CSystemCallerWin32>();
+#endif
+	return make_smart_refctd_ptr<system::ISystem>(std::move(caller));
+}
+
 int main()
 {
+	const char* APP_NAME = "01.HelloWorld";
 	constexpr uint32_t WIN_W = 800u;
 	constexpr uint32_t WIN_H = 600u;
-	constexpr uint32_t SC_IMG_COUNT = 3u; // problematic, shouldn't fix the number of swapchain images at compile time, since Vulkan is under no obligation to return you the exact number of images you requested
-	
-	// Note(achal): This is unused, for now
-	video::SDebugCallback dbgcb;
-	dbgcb.callback = &debugCallback;
-	dbgcb.userData = nullptr;
-	
-	auto system = CommonAPI::createSystem(); // Todo(achal): Need to get rid of this
+	constexpr uint32_t MAX_SWAPCHAIN_IMAGE_COUNT = 8u;
+	constexpr uint32_t FRAMES_IN_FLIGHT = 2u;
 
+	auto system = createSystem();
+	auto logger = core::make_smart_refctd_ptr<system::CColoredStdoutLoggerWin32>();
 	auto winManager = core::make_smart_refctd_ptr<nbl::ui::CWindowManagerWin32>();
+	auto eventCallback = core::make_smart_refctd_ptr<DemoEventCallback>();
 
 	nbl::ui::IWindow::SCreationParams params;
 	params.callback = nullptr;
 	params.width = WIN_W;
 	params.height = WIN_H;
-	params.x = 0;
-	params.y = 0;
+	params.x = 100;
+	params.y = 100;
 	params.system = core::smart_refctd_ptr(system);
 	params.flags = nbl::ui::IWindow::ECF_NONE;
-	params.windowCaption = "Test Window";
-	params.callback = core::make_smart_refctd_ptr<DemoEventCallback>();
+	params.windowCaption = APP_NAME;
+	params.callback = eventCallback;
 	auto window = winManager->createWindow(std::move(params));
 
-	// Todo(achal): This, for some reason, fails for Release and RelWithDebInfo builds
-	core::smart_refctd_ptr<video::IAPIConnection> vk = video::IAPIConnection::create(std::move(system), video::EAT_VULKAN, 0, "New API Test", &dbgcb);
-#if 1
-	core::smart_refctd_ptr<video::ISurface> surface = vk->createSurface(window.get());
+	std::cout <<
+		R"(
+Choose Graphics API:
+0) Vulkan
+1) OpenGL
+2) OpenGL ES
+)" << std::endl;
+
+	int apiType;
+	std::cin >> apiType;

```

Comment:
need an alternative for android (always choose ES, or random choice between ES and Vulkan)